internal/summary/summary.go

```diff
@@ -24,29 +33,87 @@ func NewSummary() *Summary {
 }
 
 // Gather collates data into Summary
-func (s *Summary) Gather(services []model.Service) {
+func (s *Summary) Gather(services []model.Service) error {
 
 	for _, svc := range services {
-		var path string
 		// Assume that, in practice, there is only one mount
+		path := ""
 		if len(svc.Mounts) > 0 {
 			path = svc.Mounts[0].Repo.LocalPath
-		} else {
-			path = ""
 		}
-		s.Services = append(s.Services, &Service{
-			Name: string(svc.Name),
-			Path: path,
-		})
+
+		svcSummary := &Service{
+			Name:    string(svc.Name),
+			Path:    path,
+			k8sData: []k8sDatum{},
+		}
+
+		kubeDatum := &k8sDatum{}
```

Comment:
this shouldn't be a pointer if you're ultimately storing `[]k8sDatum` (i.e. a slice of NOT pointers, but the structs themselves) -- see below where you call `append(svcSummary.k8sData, *kubeDatum)`. 

if you initialize here with `kubeDatum := &k8sDatum{}` then you don't have to use the star in the `append` call below. lmk if you wanna chat more about pointers.